docs/dask.rst

```diff
@@ -53,3 +62,133 @@ a progress bar:
     >>> cube.max()  # doctest: +IGNORE_OUTPUT
     [########################################] | 100% Completed |  0.1s
     <Quantity 0.01936739 Jy / beam>
+
+Performance benefits of using dask spectral cube classes
+
+Saving intermediate results to disk
+-----------------------------------
+
+When calling methods such as for example :meth:`~spectral_cube.DaskSpectralCube.convolve_to` or any other
+methods that return a cube, the result is not immediately calculated - instead, the result is only computed
+when data is accessed directly (for example via `~spectral_cube.DaskSpectralCube.filled_data`), or when
+writing the cube to disk, for example as a FITS file. However, when doing several operations in a row, such
+as spectrally smoothing the cube, then spatially smoothing it, it can be more efficient to store intermediate
+results to disk. All methods that return a cube can therefore take the ``save_to_tmp_dir`` option which can
+be set to `True` to compute the result of the operation immediately, save it to a temporary directory, and
+re-read it immediately from disk (for users interested in how the data is stored, it is stored as a
```

Comment:
I think this might be system-dependent, I'm basically just using the Python ``tempfile`` module so I assume the system cleans this up from time to time (e.g. at reboot). I'll open an issue so that we can investigate in detail since these files will be big.

Reply:
See https://github.com/radio-astro-tools/spectral-cube/issues/633